src/gui/widgets/spinner.cpp

```diff
@@ -150,26 +221,60 @@ spinner_definition::resolution::resolution(const config& cfg)
 namespace implementation
 {
 
-builder_spinner::builder_spinner(const config& cfg)
-	: implementation::builder_styled_widget(cfg)
-{
-}
+	builder_spinner::builder_spinner(const config& cfg)
+		: implementation::builder_styled_widget(cfg)
+		  , step_size_(cfg["step_size"].to_int(1))
+	{
+		if(cfg.has_attribute("minimum_value")) {
+			minimum_value_ = cfg["minimum_value"].to_int();
```

Comment:
I think setting a default value for `maximum_value` and `minimum_value` can simplify much of the checks and pointer existence checking. I'd use 0/`std::numeric_limits<int>::min()` for a default min value and `std::numeric_limits<int>::max()` for a default max value.

Reply:
I suppose numeric_limits really are the limits, even if the user wants "no limit" on either/both.  I think you still want the user to be able to use nil in lua to represent "no limit", but I'm not sure force there to be a limit would preclude that.  

Even though I think the user will specify 0 in most cases for the min, I wouldn't want to force that on them (with WML I don't see a way for them to say "No, not 0, no limit", besides just setting a really "big" limit).  

I'll try it out and see if it turns out to be simpler.

Reply:
> with WML I don't see a way for them to say "No, not 0, no limit", besides just setting a really "big" limit

`minimum_value=infinity` or `maximum_value=infinity` should be fine, right?

Reply:
> > with WML I don't see a way for them to say "No, not 0, no limit", besides just setting a really "big" limit
> 
> `minimum_value=infinity` or `maximum_value=infinity` should be fine, right?

I don't know, is that a thing (is infinity a special value in WML, and if so how does that translate to cpp)?  Or do you mean make it them strings and handle "infinity" as a special case?

I'd probably say `minimum_value=infinity` is wrong though.  I mean, I think the 'negative' is implied, but I'm not a fan of implied when what is actually expressed means something very different.